scripts/base/protocols/ldap/main.zeek

```diff
@@ -0,0 +1,513 @@
+      }
+
+      if (( ! c$ldap_messages[message_id]?$proto ) && c?$ldap_proto)
+        c$ldap_messages[message_id]$proto = c$ldap_proto;
+
+      Log::write(LDAP::LDAP_LOG, c$ldap_messages[message_id]);
+      delete c$ldap_messages[message_id];
+    }
+  }
+
+}
+
+#############################################################################
+event LDAP::searchreq(c: connection,
+                      message_id: int,
+                      base_object: string,
+                      scope: LDAP::SearchScope,
+                      deref: LDAP::SearchDerefAlias,
+                      size_limit: int,
+                      time_limit: int,
+                      types_only: bool,
+                      filter: string,
+                      attributes: vector of string) {
+
+  set_session(c, message_id, LDAP::ProtocolOpcode_SEARCH_REQUEST);
+
+  if ( scope != LDAP::SearchScope_Undef ) {
+    if ( ! c$ldap_searches[message_id]?$scope )
+      c$ldap_searches[message_id]$scope = set();
+    add c$ldap_searches[message_id]$scope[SEARCH_SCOPES[scope]];
+  }
+
+  if ( deref != LDAP::SearchDerefAlias_Undef ) {
+    if ( ! c$ldap_searches[message_id]?$deref )
+      c$ldap_searches[message_id]$deref = set();
+    add c$ldap_searches[message_id]$deref[SEARCH_DEREF_ALIASES[deref]];
+  }
+
+  if ( base_object != "" ) {
+    if ( ! c$ldap_searches[message_id]?$base_object )
+      c$ldap_searches[message_id]$base_object = vector();
+    c$ldap_searches[message_id]$base_object += base_object;
+  }
+  c$ldap_searches[message_id]$filter = filter;
+
+  if ( default_log_search_attributes ) {
+    c$ldap_searches[message_id]$attributes = attributes;
+  }
+}
+
+#############################################################################
+event LDAP::searchres(c: connection,
+                      message_id: int,
+                      object_name: string) {
+
+  set_session(c, message_id, LDAP::ProtocolOpcode_SEARCH_RESULT_ENTRY);
+
+  c$ldap_searches[message_id]$result_count += 1;
+}
+
+#############################################################################
+event LDAP::bindreq(c: connection,
+                    message_id: int,
+                    version: int,
+                    name: string,
+                    authType: LDAP::BindAuthType,
+                    authInfo: string) {
+
+  set_session(c, message_id, LDAP::ProtocolOpcode_BIND_REQUEST);
+
+  if ( ! c$ldap_messages[message_id]?$version )
+    c$ldap_messages[message_id]$version = version;
+
+  if ( ! c$ldap_messages[message_id]?$opcode )
+    c$ldap_messages[message_id]$opcode = set();
+
+  if (authType == LDAP::BindAuthType_BIND_AUTH_SIMPLE) {
+    add c$ldap_messages[message_id]$opcode[BIND_SIMPLE];
+  } else if (authType == LDAP::BindAuthType_BIND_AUTH_SASL) {
+    add c$ldap_messages[message_id]$opcode[BIND_SASL];
+  }
+
+}
+
+#############################################################################
+event connection_state_remove(c: connection) {
```

Comment:
The conventional way is to use conn-removal hooks instead of connection_state_remove handlers (this handler is invoked also for non-ldap connections).

The package template was adapated, but I'm sure this code stems from before.